src/App.tsx

```diff
@@ -191,33 +192,29 @@ function getMessage(logging: JSON): undefined | getMessageReturn {
   }
 }
 
-function logLevel(text: string): [string, string, number] {
-  const debug_levels: {[key: string]: number} = {
-    "EMERG":0,
-    "PANIC":0,
-    "ALERT":1,
-    "CRIT":2,
-    "ERROR":3, 
-    "WARN":4, 
-    "NOTICE":5, 
-    "INFO":6,
-    "DEBUG":7,};
+function getLogLevel(text: string, level_val:number): [string, string] {
+  const log_levels: {[key: number]: string} = {
+    0:"EMERG",
+    1:"ALERT",
+    2:"CRIT",
+    3:"ERROR", 
+    4:"WARN", 
+    5:"NOTICE", 
+    6:"INFO",
+    7:"DEBUG",};
   const words = text.split(/\s+/);
   const firstWord = words[0] || '';  
   const restOfText = words.slice(2).join(' ');
-  let debug = "";
+  let level = "";
   let message = "";
-  let level_val:number = 0;
-  if (firstWord in debug_levels) {
-    debug = firstWord;
+  if (Object.values(log_levels).includes(firstWord)) {
+    level = log_levels[level_val] || "UNKNOWN";
     message = restOfText;
-    level_val = debug_levels[firstWord];
   } else {
-    debug = "UNKNOWN";
+    level = "UNKNOWN";
     message = text;
-    level_val = 7;
   }
```

Comment:
From discussion, needed because the log level is prepended to the `message` field, there is also a `full_message` field that may work and doesn't have the log level prepended, so worth a try. If that doesn't work I think this can remain as-is.